api/src/main/java/org/openmrs/module/fhir2/api/impl/FhirMedicationRequestServiceImpl.java

```diff
@@ -9,27 +9,31 @@
  */
 package org.openmrs.module.fhir2.api.impl;
 
-import lombok.AccessLevel;
-import lombok.Setter;
 import org.hl7.fhir.r4.model.MedicationRequest;
 import org.openmrs.module.fhir2.api.FhirMedicationRequestService;
 import org.openmrs.module.fhir2.api.dao.FhirMedicationRequestDao;
 import org.openmrs.module.fhir2.api.translators.MedicationRequestTranslator;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.Setter;
+
 @Component
 @Setter(AccessLevel.PACKAGE)
-public class FhirMedicationRequestServiceImpl implements FhirMedicationRequestService {
+@Getter(AccessLevel.PROTECTED)
+public class FhirMedicationRequestServiceImpl extends BaseFhirService<MedicationRequest, org.openmrs.DrugOrder> implements FhirMedicationRequestService {
 	
 	@Autowired
-	private MedicationRequestTranslator medicationRequestTranslator;
+	private MedicationRequestTranslator translator;
 	
 	@Autowired
 	private FhirMedicationRequestDao dao;
-	
-	@Override
-	public MedicationRequest getMedicationRequestByUuid(String uuid) {
-		return medicationRequestTranslator.toFhirResource(dao.getMedicationRequestByUuid(uuid));
+
+	public void setMedicationRequestTranslator(MedicationRequestTranslator translator) {
+		this.translator = translator;
 	}
```

Comment:
This also shouldn't be necessary. You can tweak the AccessLevel of the `@Setter` annotation above if this needs to be protected or public.

Reply:
@ibacher the challange is that i did tweak the @setter annotation but for some reason the compiler still complains that the setMedicationRequestTranslator isn't defined in `FhirMedicationRequestServiceImpl`, thats why am forced to define it here

Reply:
@gitcliff In the unit tests, make sure you've changed them to call `setTranslator()` instead of `setMedicationRequestTranslator()`. That should eliminate the complaints. 😄

Reply:
sure ,,did that and its clear now